File: Net-Update.cs
using System.Net.Http;
using System;

namespace NetUpdate
{
    public static class NetUpdate
    {
        private static readonly HttpClient client = new HttpClient();
        static void update(string appName, string appVer, string rawGistURL)
        {
            try
            {
                client.Timeout = TimeSpan.FromSeconds(5);
                HttpResponseMessage response = client.GetAsync(rawGistURL).GetAwaiter().GetResult();
                response.EnsureSuccessStatusCode();
                string latestVersion = response.Content.ReadAsStringAsync().GetAwaiter().GetResult().Trim();
                Version current = new Version(appVer);
                Version latest = new Version(latestVersion);

                if (latest.CompareTo(current) > 0)
                {
                    Console.WriteLine($"Woohoo! A NEW version of {appName} is Available!\n");                    
                }
                else if (latest.CompareTo(current) == 0)
                {
                    Console.WriteLine($"Hurrah! {appName} is up to date!\n");
                }
                else
                {
                    Console.WriteLine("WARNING! This appears to be a DEV. Build.\n");
                }
            }
            catch (HttpRequestException e)
            {
                Console.WriteLine("Could not check for updates. Please check your internet connection.");
                Console.WriteLine($"Error: {e.Message}\n");
            }
            catch (TaskCanceledException ex) when (ex.InnerException is TimeoutException)
            {
                Console.WriteLine("Update check timed out. Please check your internet connection.\n");
            }
            catch (FormatException)
            {
                Console.WriteLine("Could not parse the version from the update source.\n");
            }
            catch (Exception e)
            {
                Console.WriteLine("An unexpected error occurred during the update check.");
                Console.WriteLine($"Error: {e.Message}\n");
            }
        }
    }
}

File: Net_Update.py
import requests

class UpdateCheck : # Net-Update v1.0 by Chill-Astro
    @staticmethod
    def update(appVer, appName, gistURL) :
        try :
            response = requests.get(gistURL, timeout=5) # Fetch version file, timeout after 5 seconds
            response.raise_for_status()  # Raise an error for bad responses
            newVer = response.text.strip()  # Get the version from the response
            if newVer > appVer :
                print(f"🎉 A NEW version of {appName} is Available! : {newVer}\n")
            elif newVer == appVer :
                print(f"🎉 {appName} is up to date!\n") # Latest Version
            elif newVer < appVer :
                print(f"⚠️  This is a DEV. Build of {appName}!\n") # For development builds, we assume the version is lower
        except requests.RequestException as e:
             print("⚠️  Could not check for updates. Please check your internet connection.\n" f"Error: {e}\n")
        except Exception as e: # Catching other potential errors, like if the content of the file is not proper.
            print("⚠️  Error occurred while checking for updates.\n" f"Error: {e}\n")